src/Features/Core/Portable/ReplaceDocCommentTextWithTag/AbstractReplaceDocCommentTextWithTagCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,226 @@
+            {
+                return;
+            }
+
+            if (!token.FullSpan.Contains(span))
+            {
+                return;
+            }
+
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+            var singleWordSpan = ExpandSpan(sourceText, span, fullyQualifiedName: false);
+            var singleWordText = sourceText.ToString(singleWordSpan);
+            if (singleWordText == "")
+            {
+                return;
+            }
+
+            // First see if they're on a keyword. 
+            if (IsAnyKeyword(singleWordText))
+            {
+                RegisterRefactoring(context, singleWordSpan, $@"<see langword=""{singleWordText}""/>");
+                return;
+            }
+
+            // Not a keyword, see if it semantically means anything in the current context.
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var symbol = GetEnclosingSymbol(semanticModel, span.Start, cancellationToken);
+            if (symbol == null)
+            {
+                return;
+            }
+
+            // See if we can expand the term out to a fully qualified name. Do this
+            // first in case the user has something like X.memberName.  We don't want 
+            // to try to bind "memberName" first as it might bind to something like 
+            // a parameter, which is not what the user intends
+            var fullyQualifiedSpan = ExpandSpan(sourceText, span, fullyQualifiedName: true);
+            if (fullyQualifiedSpan != singleWordSpan)
+            {
+                var fullyQualifiedText = sourceText.ToString(fullyQualifiedSpan);
+                if (TryParseAndBindText(context, semanticModel, token, fullyQualifiedSpan, fullyQualifiedText))
+                {
+                    return;
+                }
+            }
+
+            // Check if the single word could be binding to a type parameter or parameter
+            // for the current symbol.
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var parameter = symbol.GetParameters().FirstOrDefault(p => syntaxFacts.StringComparer.Equals(p.Name, singleWordText));
+            if (parameter != null)
+            {
+                RegisterRefactoring(context, singleWordSpan, $@"<paramref name=""{singleWordText}""/>");
+                return;
+            }
+
+            var typeParameter = symbol.GetTypeParameters().FirstOrDefault(t => syntaxFacts.StringComparer.Equals(t.Name, singleWordText));
+            if (typeParameter != null)
+            {
+                RegisterRefactoring(context, singleWordSpan, $@"<typeparamref name=""{singleWordText}""/>");
+                return;
+            }
+
+            // Look up the term inside the named type if we're inside of one.
+            if (symbol is INamedTypeSymbol namedType)
```

Comment:
it's more for correctness.  When looking up a name, the compiler looks up from inside the type firs,t seeing members, before seeing outer symbols.